pkg/agent/storage/provider.go

```diff
@@ -29,9 +29,11 @@ const (
 	// File  Protocol = "file://"
 	HTTPS Protocol = "https://"
 	HTTP  Protocol = "http://"
+	// OCI dist spec v1.1.0
+	OCI Protocol = "oci://"
```

Comment:
Modelcars also use `oci://` as a schema prefix, so that might be conflicting here. Modelcars in fact, are conventional OCI images as they have also an active component for keeping a container alive (notably `sleep` and also a `ln -s` for creating the symbolic link in the emptyDir volume to point to the model data).

I'm not sure how the PR proposed here infers with the logic of the modelcar as defined in the storage initializer.

Reply:
The images here are going to be stored from a OCI artifact registry, so indeed they are also oci images. The question then is the transport schema. Maybe "oci-registry://" similar to "model-registry://"

Reply:
@rchincha sorry, I'm not deep in this PR, but how do you plan to make the modeldata in the OCI image accessible to the runtime ? For me it looks like that the existing [modelcar feature](https://kserve.github.io/website/latest/modelserving/storage/oci/) is already quite similar what you are trying to achieve with this PR. It would be cool, if we could merge your work with that.

If not, what would be the difference to https://kserve.github.io/website/latest/modelserving/storage/oci/ ? (just for my understanding)

Reply:
@rhuss indeed, the "effect" is the same, except the modelcar approach needs a running container whereas this PR is downloading model data and purely a storage approach. Also my slow progress on this PR as I am trying to understand more. No question that the modelcar approach although one can argue is a "hack", solves lots of pertinent issues and very effective.

Just FYI that there are some discussions happening in the opencontainers (OCI) side about this use case - pulling large things and making them available to container workloads.

Reply:
I agree, I see modelcars also only as a temporary solution until a proper OCI image volume mount type is available for Kubernetes (see also KEP https://github.com/kubernetes/enhancements/pull/4642) 

I haven't yet looked into your PR. Still, I'm really curious how direct OCI image access could work today (as many have tried but not really succeeded, except via that `shareProcessNamespace` trick, if you follow the long list of comments on https://github.com/kubernetes/kubernetes/issues/831), so I will definitely have a look. 

Let's check how we can combine our efforts to cover the current state while being prepared for future Kubernetes improvements (especially when it comes to which schema to select).

Reply:
Also to be aware of, fyi ...

https://github.com/opencontainers/distribution-spec/issues/536
https://github.com/opencontainers/image-spec/issues/1190

Reply:
FYI, https://github.com/opencontainers/distribution-spec/pull/537 has been merged with some data points.

Reply:
https://github.com/opencontainers/distribution-spec/issues/494
^ introducing blake3 helps with large artifact generation and verification, because blake3 thanks to merkle structure is very fast.